proposals/0000-Move-Platforms-to-Own-Packages.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Move iOS and Android Specific Code to Their Own Packages
+author:
+- Ian Meikle
+date: today
+---
+
+# RFC0000: Move iOS and Android Specific Code to Their Own Packages
+
+## Summary
+
+Re-organize the React Native repo as a Yarn workspace monorepo, and move platform specific JS and native code for iOS and Android into their own packages.
+
+## Basic example
+
+Common APIs and components would remain in `react-native`:
```

Comment:
How Windows and other platforms would provide their implementations of core components and APIs? Since imports are from `react-native`, not from platform-specific module, I guess that would be tricky and require certain platforms to export a minimal subset of APIs to be considered a valid platform for React Native?

Reply:
Platforms can have an entry like this in their `package.json` which adds their platform to the haste search path:

```json
{
  "rnpm": {
    "haste": {
      "providesModuleNodeModules": ["react-native-example"],
      "platforms": ["example"]
    }
  }
}
```

And if they wanted to override `ExampleComponent` in `react-native`, `react-native-example` would have `ExampleComponent.example.js`

This solution depends on Haste, so if RN wants to drop Haste in the future a solution for platform overrides will need to be worked out. 

> I guess that would be tricky and require certain platforms to export a minimal subset of APIs to be considered a valid platform for React Native

Pretty much, or have a whole bunch of stub JS `UnimplementedView` files to address the Haste missing files errors